### PR TITLE
Improvements to issue create command

### LINKED ISSUE #70: Improvements to issue create command
Created by: Living `(563138570953687061)`

Emoji to delete suggestion, and allow no description, along with better invalid commands checking

### PR DESCRIPTION
**Pull request by sourTaste000**
closes #70

### CODE REVIEW

File: src/main/kotlin/commands/IssueCommand.kt [hunk 1]
```diff
@@ -81,10 +81,10 @@ object IssueCommand : Command("issue") {
                         doesLater { context ->
                             val repo: String = context arg "repo"
                             val title: String = context arg "title"
-                            val body: String = context arg "body"
+                            val body: String? = context arg "body"
```
Comment (l1ving) [inline src/main/kotlin/commands/IssueCommand.kt:84]: why is this nullable?

File: src/main/kotlin/commands/IssueCommand.kt [hunk 2]
```diff
@@ -120,6 +120,7 @@ object IssueCommand : Command("issue") {
 
                             delay(1000)
                             form.addReaction('✅')
+                            form.addReaction('❎')
```
Comment (l1ving) [inline src/main/kotlin/commands/IssueCommand.kt:123]: isn't this supposed to be red? or is that just my phone

File: src/main/kotlin/commands/IssueCommand.kt [hunk 3]
```diff
@@ -151,19 +152,29 @@ object IssueCommand : Command("issue") {
             return
         }
 
+        if (event.reaction.emoji.name == "❎") {
```
Comment (l1ving) [inline src/main/kotlin/commands/IssueCommand.kt:155]: use an else if please

File: src/main/kotlin/commands/IssueCommand.kt [hunk 4]
```diff
@@ -151,19 +152,29 @@ object IssueCommand : Command("issue") {
...
+            try {
+                message = message.error("Rejected")
```
Comment (l1ving) [inline src/main/kotlin/commands/IssueCommand.kt:157]: leave more information, such as "Rejected issue \"$title\""

File: src/main/kotlin/commands/IssueCommand.kt [hunk 5]
```diff
@@ -151,19 +152,29 @@ object IssueCommand : Command("issue") {
...
+
+                form.first.delete()
+
+                delay(10000)
```
Comment (l1ving) [inline src/main/kotlin/commands/IssueCommand.kt:161]: 10 seconds is more than enough, do 5

File: src/main/kotlin/commands/IssueCommand.kt [hunk 6]
```diff
@@ -151,19 +152,29 @@ object IssueCommand : Command("issue") {
...
+            } catch (e: IllegalStateException) {
+                // ignored
+            }
+        }
+
         createGithubIssue(form.second, user, form.third, token)
 
         try {
             form.first.delete()
-        } catch (e: IllegalStateException) {
```
Comment (l1ving) [inline src/main/kotlin/commands/IssueCommand.kt:158]: these were intentionally individual, but now that commands have error handling you can just remove all of these

**Review by l1ving — CHANGES_REQUESTED** [review]

File: src/main/kotlin/commands/IssueCommand.kt [hunk 7]
```diff
@@ -112,14 +112,11 @@ object IssueCommand : Command("issue") {
...
-                            try {
-                                message.delete()
-                            } catch (e: IllegalStateException) {
-                                // this is fine. it just means the member list isn't cached and we can't delete it
-                            }
+                            message.delete()
 
                             delay(1000)
                             form.addReaction('✅')
+                            form.addReaction('❌')
```
Comment (l1ving) [inline src/main/kotlin/commands/IssueCommand.kt:119]: My bad, please use `❎`

https://emojipedia.org/check-mark-button/
https://emojipedia.org/cross-mark-button/

File: src/main/kotlin/commands/IssueCommand.kt [hunk 8]
```diff
@@ -81,10 +81,10 @@ object IssueCommand : Command("issue") {
                         doesLater { context ->
                             val repo: String = context arg "repo"
                             val title: String = context arg "title"
-                            val body: String = context arg "body"
+                            val body: String? = context arg "body"
 
                             val issue = Issue()
-                            val formattedIssue = "Created by: ${message.author?.name?.toHumanReadable()} `(${message.author?.id})`\n\n$body"
+                            val formattedIssue = "Created by: ${message.author?.name?.toHumanReadable()} `(${message.author?.id})`\n\n${body ?: "No description provided."}"
```
Comment (l1ving) [inline src/main/kotlin/commands/IssueCommand.kt:87]: Brigadier doesn't allow body to ever be null

File: src/main/kotlin/commands/IssueCommand.kt [hunk 9]
```diff
@@ -151,23 +148,19 @@ object IssueCommand : Command("issue") {
...
-        createGithubIssue(form.second, user, form.third, token)
-
-        try {
+        if (event.reaction.emoji.name == "❌") {
+            message = message.error("Rejected issue ${form.second.title}!")
             form.first.delete()
-        } catch (e: IllegalStateException) {
-            // this is fine. it just means the member list isn't cached and we can't delete it
+            delay(5000)
+            message.delete()
```
Comment (l1ving) [inline src/main/kotlin/commands/IssueCommand.kt:155]: you're missing a return here

**Review by l1ving — CHANGES_REQUESTED** [review]
>         if (event.reaction.emoji.name != "✅") return

As I said earlier, this should be an if else, not an if return

**Comment by l1ving** [comment]
Looks good, I'll fix a code meme myself tomorrow

**Comment by sourTaste000** [comment]
s**t wrong branch

**Comment by sourTaste000** [comment]
LOL so many conflict

**Comment by l1ving** [comment]
i fix :kannagun:

**Review by l1ving — CHANGES_REQUESTED** [review]
Sorry, your commit history is out of alignment, I just reimplemented this myself in ca08fe13cc041c78bcde8adc4c382b861523d0a7

**Pull request closed without merging**